plugins/source/slack/client/services/slack.go

```diff
@@ -0,0 +1,55 @@
+// Code generated by codegen; DO NOT EDIT.
+package services
+
+import (
+	"context"
+	"github.com/slack-go/slack"
+	"io"
+)
+
+//go:generate mockgen -package=mocks -destination=../mocks/slack.go -source=slack.go SlackClient
+type SlackClient interface {
```

Comment:
I considered this approach, but I don't think we should do it here. It requires the tests to have internal knowledge of the underlying Slack APIs, when the Slack SDK is supposed to be a wrapper around that.

Reply:
Agree with @hermanschaaf, I opened a related pitfall in HTTP server tests in https://github.com/cloudquery/cloudquery/issues/5656 (where we use a wildcard to avoid needing to figure out the correct URL/arguments)

Reply:
We err on the side of testing via http router if we can do it via SDK options.
cc: @yevgenypats